src/core/include/openvino/core/model.hpp

```diff
@@ -346,6 +346,9 @@ class OPENVINO_API Model : public std::enable_shared_from_this<Model> {
     // node has no consumers but still exists in a graph.
     mutable std::vector<std::weak_ptr<Node>> m_cached_ordered_ops;
```

Comment:
Can we fill unordered maps together with m_cached_ordered_ops?

Do you have any performance measurements?

Reply:
I didn't test all possible models from model zoo.
I tested this on synthesized models with 2k-20k nodes. And used 'add_output' to each node. Without the improvement, 1k takes ~1.7 seconds, 10k takes > 200 seconds.
With improvement, all these are completed within less than one second: 10k with O(N) is processed within same time as 100 nodes with O(N*N).

Test that I added verifies performance N1=N and N2=50*N nodes (adding results to each node) and it verifies that `T1*2500` is significantly less than `T2`